net_windows.go

```diff
@@ -0,0 +1,65 @@
+// Use and distribution licensed under the Apache license version 2.
+//
+// See the COPYING file in the root project directory for full text.
+//
+
+package ghw
+
+import (
+	"strings"
+
+	"github.com/StackExchange/wmi"
+)
+
+const wqlNetworkAdapter = "SELECT Description, DeviceID, Index, InterfaceIndex, MACAddress, Manufacturer, Name, NetConnectionID, ProductName, ServiceName  FROM Win32_NetworkAdapter"
+
+type win32NetworkAdapter struct {
+	Description     string
+	DeviceID        string
+	Index           uint32
+	InterfaceIndex  uint32
+	MACAddress      string
+	Manufacturer    string
+	Name            string
+	NetConnectionID string
+	ProductName     string
+	ServiceName     string
+}
+
+func (ctx *context) netFillInfo(info *NetworkInfo) error {
+	// Getting info from WMI
+	var win32NetDescriptions []win32NetworkAdapter
+	if err := wmi.Query(wqlNetworkAdapter, &win32NetDescriptions); err != nil {
+		return err
+	}
+
+	info.NICs = ctx.nics(win32NetDescriptions)
+	return nil
+}
+
+func (ctx *context) nics(win32NetDescriptions []win32NetworkAdapter) []*NIC {
+	// Converting into standard structures
+	nics := make([]*NIC, 0)
+	for _, nicDescription := range win32NetDescriptions {
+		nic := &NIC{
+			Name:         ctx.netDeviceName(nicDescription),
+			MacAddress:   nicDescription.MACAddress,
+			IsVirtual:    false,
+			Capabilities: []*NICCapability{},
+		}
+		// Appenging NIC to NICs
+		nics = append(nics, nic)
+	}
+
+	return nics
+}
+
+func (ctx *context) netDeviceName(description win32NetworkAdapter) string {
+	var name string
+	if strings.TrimSpace(description.NetConnectionID) != "" {
+		name = description.NetConnectionID + " - " + description.Description
```

Comment:
I think it would probably be better to just use `NetConnectionID` here and not append the `Description` field if `NetConnectionID` is non-null. Otherwise it's likely you'll end up with NICs named "Wired Connection - Wired Connection" or something like that..

Reply:
I was at the same idea at first but during some tests on the field I found out that their union was more usefull (and sometimes, even mandatory).

Here a quick example on Windows Machine using the command **"go run .\cmd\ghwc\main.go -f yaml net"**:

![image](https://user-images.githubusercontent.com/886518/74085468-7bfe8f00-4a79-11ea-9563-a2bce72be610.png)

As you can see, their union often deepens the description of the device.

The Latins said: "[Melius abundare quam deficere](https://en.wikipedia.org/wiki/List_of_Latin_phrases_(M)#Melius_abundare)"
We could leave their union in the "name" field or 
hypothesize a new field.

Reply:
:) OK, you'd convinced me, Riccardo!